src/generic/Map.jl

```diff
@@ -19,20 +19,35 @@ function (f::CompositeMap{D, C})(a) where {D, C}
    return f.map2(f.map1(a))::elem_type(C)
 end
 
-function show_short(io::IO, M::CompositeMap)
-   show_short(io, M.map1)
-   println(io, "then")
-   print(io, M.map2)
+function show(io::IO, ::MIME"text/plain", M::CompositeMap)
+   io = pretty(io)
+   println(io, "Composite map")
+   println(io, Indent(), "first ", Lowercase(), map1(M))
+   h = map2(M)
+   while h isa FunctionalCompositeMap || h isa CompositeMap
+     println(io, "next ", Lowercase(), h)
+     h = map2(h)
+   end
+   print(io, "next ", Lowercase(), h, Dedent())
 end
 
 function show(io::IO, M::CompositeMap)
-   println(io, "Composite map consisting of the following")
-   println(io, "")
-   show_short(io, M.map1)
-   println(io, "then")
-   show_short(io, M.map2)
+   if get(io, :supercompact, false)
+      # no nested printing
+      print(io, "Composite map")
+   else
+     io = pretty(io)
+      print(io, "Map: ", Lowercase(), domain(M))
+      h = map2(m)
```

Comment:
Tests are good, even for "obvious" code, as they uncover bugs like this one:
```suggestion
      h = map2(M)
```
I'll push a fix and adjusted doctests in a moment. Since PR #1431 seems to have stalled :-( I'd rather get this PR here ready ASAP.